docs: Document differences between MBTiles and PMTiles (#2694)
maplibre/martin: docs/content/sources-files.md

    We also don't currently support refreshing the catalog at runtime.
    If you want to implement this feature, please see <https://github.com/maplibre/martin/issues/288> instead.

## MBtiles vs PMTiles

The difference between MBTiles and PMTiles is that:

- **MBTiles** require the entire archive to be on the same machine. **PMTiles** can utilise a remote HTTP-Range request supporting server or a local file.
- Performance wise, **MBTiles** is slightly faster than **PMTiles**, but with caching this is negligible.
- Disk size wise, **MBTiles** is slightly (10-15%) higher than **PMTiles**.
- **PMTiles** requires less memory in extreme cases as sqlite has a small in-memory cache.

PMtiles can also be served **without martin**, but this has a few caveats:

- If not set up separately and used maliciously, being able to connect and download the planet could incurr network costs on some cloud providers.
- PMTiles sligtly overfetches each tile to avoid needing to do multiple requests

> - Cannot be mixed with dynamic datasources (=PostGIS) as simple

Author: ```suggestion
- Cannot be mixed with dynamic datasources (=PostGIS) as simple
- Caching is likely less effective
```

    We also don't currently support refreshing the catalog at runtime.
    If you want to implement this feature, please see <https://github.com/maplibre/martin/issues/288> instead.

## MBtiles vs PMTiles

The difference between MBTiles and PMTiles is that:

- **MBTiles** require the entire archive to be on the same machine. **PMTiles** can utilise a remote HTTP-Range request supporting server or a local file.
- Performance wise, **MBTiles** is slightly faster than **PMTiles**, but with caching this is negligible.
- Disk size wise, **MBTiles** is slightly (10-15%) higher than **PMTiles**.
- **PMTiles** requires less memory in extreme cases as sqlite has a small in-memory cache.

PMtiles can also be served **without martin**, but this has a few caveats:

- If not set up separately and used maliciously, being able to connect and download the planet could incurr network costs on some cloud providers.
- PMTiles sligtly overfetches each tile to avoid needing to do multiple requests
- Cannot be mixed with dynamic datasources (=PostGIS) as simple

The choice depends on your specific usecase and requirements.

> 

Author: ```suggestion
- Cannot be mixed with dynamic datasources (=PostGIS) as simple

The choice depends on your specific usecase and requirements.

```

> The choice depends on your specific usecase and requirements.

Author: ```suggestion
## MBTiles vs PMTiles

MBTiles and PMTiles are both formats for storing tiled geospatial data, but they differ in architecture, deployment, and operational characteristics.

### Key Differences

* **Deployment model**
  **MBTiles** archives must be stored locally on the same machine as the tile server.
  **PMTiles** archives can be accessed either locally or remotely via HTTP range requests, enabling direct use from object storage (e.g., S3-compatible systems).
* **Performance**
  **MBTiles** typically provide slightly lower latency due to local access and SQLite indexing.
  **PMTiles** may introduce additional latency when accessed remotely, but this is usually mitigated by HTTP caching and CDN usage.
* **Storage efficiency**
  **PMTiles** archives are generally more space-efficient, typically ~10–15% smaller than equivalent **MBTiles** archives.
* **Memory usage**
  **MBTiles** relies on SQLite, which maintains an internal page cache and may increase memory usage under load.
  **PMTiles** can, in some cases, operate with lower memory overhead, depending on access patterns and caching configuration.

### Serving PMTiles without a Tile Server

PMTiles archives can be served directly from HTTP range–capable storage without a dedicated tile server. This approach has several limitations:

* **Unrestricted access risk**
  Without proper access controls, clients may download large portions (or all) of an archive, potentially incurring significant network egress costs.
* **Over-fetching**
  PMTiles may fetch more data than strictly required per tile request to minimize the number of HTTP requests.
* **Lack of source composition**
  Direct serving does not support combining PMTiles with dynamic data sources (e.g., PostGIS) into a unified tile service.
  A tile server (e.g, Martin) is required for this.
* **Caching behavior**
  Cache efficiency may be reduced compared to setups with a dedicated tile server that can optimize request patterns.

The choice between MBTiles and PMTiles depends on system requirements:

* Use **MBTiles** for local, self-contained deployments with minimal external dependencies.
* Use **PMTiles** for cloud-native or distributed setups where remote access, CDN integration, or object storage is preferred.

```

    We also don't currently support refreshing the catalog at runtime.
    If you want to implement this feature, please see <https://github.com/maplibre/martin/issues/288> instead.

## MBTiles vs PMTiles

MBTiles and PMTiles are both formats for storing tiled geospatial data, but they differ in architecture, deployment, and operational characteristics.

### Key Differences

* **Deployment model**
  **MBTiles** archives must be stored locally on the same machine as the tile server.

>   **PMTiles** archives can be accessed either locally or remotely via HTTP range requests, enabling direct use from object storage (e.g., S3-compatible systems).

Reviewer: ```suggestion
  **PMTiles** archives can be accessed either locally or remotely via HTTP range requests, e.g. from an object storage like S3.
  **PMTiles** allows simpler production deployment with Kubernetes, as it allows the large data file to reside in S3 and shared by multiple pods, but restricted from direct user access.
```

* **Performance**
  **MBTiles** typically provide slightly lower latency due to local access and SQLite indexing.
  **PMTiles** may introduce additional latency when accessed remotely, but this is usually mitigated by HTTP caching and CDN usage.
* **Storage efficiency**
  **PMTiles** archives are generally more space-efficient, typically ~10–15% smaller than equivalent **MBTiles** archives.
* **Memory usage**
  **MBTiles** relies on SQLite, which maintains an internal page cache and may increase memory usage under load.
  **PMTiles** can, in some cases, operate with lower memory overhead, depending on access patterns and caching configuration.

### Serving PMTiles without a Tile Server

PMTiles archives can be served directly from HTTP range–capable storage without a dedicated tile server. This approach has several limitations:

* **Unrestricted access risk**

>   Without proper access controls, clients may download large portions (or all) of an archive, potentially incurring significant network egress costs.

Reviewer: i don't think S3 allows any kind of range-request size restrictions like this, but i guess this is fine

Author: I am thinking about a gateway in frot, so API Gateway (?) or one of the 3-4 other reverse proxy/LB solutions.

```suggestion
  Without proper access controls, clients may download large portions (or all) of an archive, leading to significant egress costs.
  A tile server restricts access to tile requests, but bulk extraction remains possible via many requests, which are generally easier to detect and block.
```

### Serving PMTiles without a Tile Server

PMTiles archives can be served directly from HTTP range–capable storage without a dedicated tile server. This approach has several limitations:

* **Unrestricted access risk**
  Without proper access controls, clients may download large portions (or all) of an archive, potentially incurring significant network egress costs.
* **Over-fetching**
  PMTiles may fetch more data than strictly required per tile request to minimize the number of HTTP requests.
* **Lack of source composition**
  Direct serving does not support combining PMTiles with dynamic data sources (e.g., PostGIS) into a unified tile service.
  A tile server (e.g, Martin) is required for this.
* **Caching behavior**
  Cache efficiency may be reduced compared to setups with a dedicated tile server that can optimize request patterns.

The choice between MBTiles and PMTiles depends on system requirements:

* Use **MBTiles** for local, self-contained deployments with minimal external dependencies.
* Use **PMTiles** for cloud-native or distributed setups where remote access, CDN integration, or object storage is preferred.

> 
> 
> ### Serving PMTiles from local file systems, http or Object Storage

Author: ```suggestion
* Use **PMTiles** for cloud-native or distributed setups where remote access, CDN integration, or object storage is preferred.

### Serving PMTiles from local file systems, http or Object Storage
```
